src/components/ContactDetails/ContactDetailsAvatar.vue

```diff
@@ -357,13 +354,18 @@ export default {
 
 				this.loading = true
 				try {
-					const { get } = await axios()
-					const response = await get(`${imageUrl}`, {
+					const response = await axios.get(`${imageUrl}`, {
 						responseType: 'arraybuffer',
 					})
 					if (response.status !== 200) { throw new URIError('verify social profile id') }
 					OC.Notification.showTemporary(t('contacts', 'Image updated.'))
-					this.$store.dispatch('updateContact', this.contact) // FIXME: refresh not working
+					// refresh view
+					// FIXME: not working
+					const key = contactId + '~' + addressbookId
+					const updated = this.$store.getters.getContact(key)
+					this.$store.dispatch('fetchFullContact', { contact: updated })
+					this.$store.dispatch('updateContact', updated)
```

Comment:
updateContact is to send a contact to the server, that is not what you want :)
What you want is `updateLocalContact`

```suggestion
					this.$emit('updateLocalContact', updated)
```

Then https://github.com/nextcloud/contacts/blob/df9e046a18dd0871e2c18ea0bc8abb8ce0872c2f/src/components/ContactDetails.vue#L42
`<ContactAvatar :contact="contact" @updateLocalContact="updateLocalContact" />`